.github/workflows/cd.yml

```diff
@@ -1,13 +1,19 @@
 name: cd-dev
 
+permissions:
+  id-token: read
+
 on:
   push:
     branches:
       - master
+  pull_request:
+    branches:
+      - master
```

Comment:
if you need this for testing, please use a workflow_dispatch event instead, so you don't have to remember to remove the trigger afterwards